app/controllers/admin/organizations_controller.rb

```diff
@@ -32,23 +32,26 @@ def new
     @organization = Organization.new
     account_request = params[:token] && AccountRequest.get_by_identity_token(params[:token])
 
-    if account_request.blank?
-      @organization.users.build(organization_admin: true)
-    elsif account_request.processed?
+    @user = User.new
+    return unless account_request
+
+    if account_request.processed?
       flash[:error] = "The account request had already been processed and cannot be used again"
-      @organization.users.build(organization_admin: true)
     else
       @organization.assign_attributes_from_account_request(account_request)
+      @user.assign_attributes(email: account_request.email, name: account_request.name)
     end
   end
 
   def create
     @organization = Organization.new(organization_params)
-    @organization.users.last.assign_attributes(password: SecureRandom.uuid)
 
     if @organization.save
       Organization.seed_items(@organization)
-      @organization.users.last.invite!
+      @user = UserInviteService.invite(name: user_params[:name],
+                                       email: user_params[:email],
+                                       roles: %i(org_user org_admin),
```

Comment:
Perhaps we can create a class or module that keeps the constant names of the roles?

Reply:
Yep, I can do this.

Reply:
Yes, I think we do. We have both org and partner users, and we have both org admins and super admins.

Reply:
Makes sense. I think specifically the super_admin and difference between org admins make sense here.